### PR TITLE
[RISCV] Separate HW/SW shadow stack on RISC-V

### PR DESCRIPTION
**Pull request by jaidTw**
This patch follows https://github.com/llvm/llvm-project/pull/112477.
Previously `-fsanitize=shadow-call-stack` (which get transform to `Attribute::ShadowCallStack`) is used for enable both hardware and software shadow stack, and another option `-force-sw-shadow-stack` is needed if the user wants to use the software shadow stack where hardware software shadow stack could be supported. It decouples both by using the string attribute `hw-shadow-stack` to distinguish from the software shadow stack attribute.

### CODE REVIEW

**Comment by jaidTw** [comment]
@kito-cheng Do you think it's appropriate to change the dependence of shadow stack related features to Zimop at this point?

File: llvm/lib/Target/RISCV/RISCVFrameLowering.cpp [hunk 1]
```diff
@@ -73,7 +77,7 @@ static void emitSCSPrologue(MachineFunction &MF, MachineBasicBlock &MBB,
     return;
 
   const RISCVInstrInfo *TII = STI.getInstrInfo();
-  if (!STI.hasForcedSWShadowStack() && STI.hasStdExtZicfiss()) {
+  if (HasHWShadowStack) {
     BuildMI(MBB, MI, DL, TII->get(RISCV::SSPUSH)).addReg(RAReg);
```
[the remark this answers is not in the document]
Comment (jaidTw) [reply llvm/lib/Target/RISCV/RISCVFrameLowering.cpp:81]: I guess Kito is too busy, so I just choose to loosen the requirement in tablegen to Zimop

[the remark this answers is not in the document]
Comment (mylai-mtk) [reply llvm/lib/Target/RISCV/RISCVFrameLowering.cpp:81]: Actually I ran into a similar problem with my other patch, and kito expressed that they would try to accommodate this at <https://github.com/llvm/llvm-project/pull/109600#issuecomment-2375827448> if I understand it correctly ("Craig and me will fix that on the back-end side :P").

Not sure how's their progress though.

**Review by kito-cheng — APPROVED** [review]
LGTM

**Comment by mylai-mtk** [comment]
LGTM

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**